internal/resource-request-operator/suite_test.go

```diff
@@ -103,13 +113,100 @@ var _ = BeforeSuite(func(done Done) {
 
 	k8sClient = k8sManager.GetClient()
 	Expect(k8sClient).ToNot(BeNil())
+
+	ctx = context.TODO()
+	resources = corev1.ResourceList{}
+	resources[corev1.ResourceCPU] = *resource.NewQuantity(2, resource.DecimalSI)
+	resources[corev1.ResourceMemory] = *resource.NewQuantity(1, resource.BinarySI)
+	resources[corev1.ResourceLimitsCPU] = *resource.NewQuantity(3, resource.DecimalSI)
+	node1 = &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-node1",
+		},
+		Status: corev1.NodeStatus{
+			Capacity:    resources,
+			Allocatable: resources,
+			Phase:       corev1.NodeRunning,
+		},
+	}
+	node2 = &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-node2",
+		},
+		Status: corev1.NodeStatus{
+			Capacity:    resources,
+			Allocatable: resources,
+			Phase:       corev1.NodeRunning,
+		},
+	}
+	pod1 = &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-pod1",
+			Labels: map[string]string{
+				forge.LiqoOutgoingKey:     "test",
+				forge.LiqoOriginClusterID: "2468825c-0f62-44d7-bed1-9a7bc331c0b0",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				0: {
+					Name: "test-container1",
+					Resources: corev1.ResourceRequirements{
+						Limits:   resources,
+						Requests: resources,
+					},
+				},
+
+				1: {
+					Name: "test-container2",
+					Resources: corev1.ResourceRequirements{
+						Limits:   resources,
+						Requests: resources,
+					},
+				},
+			},
+		},
+	}
+
+	wrongPod = &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-wrongPod",
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				0: {
+					Name: "test-container1",
+					Resources: corev1.ResourceRequirements{
+						Limits:   resources,
+						Requests: resources,
+					},
+				},
+
+				1: {
+					Name: "test-container2",
+					Resources: corev1.ResourceRequirements{
+						Limits:   resources,
+						Requests: resources,
+					},
+				},
+			},
+		},
+	}
+	_, err = clientset.CoreV1().Nodes().Create(ctx, node1, metav1.CreateOptions{})
```

Comment:
The test-related resources should not be global variables and they should be defined inside the test-case

In the beforeEach you can create the cluster, in the afterEach delete it, and in each It case create the specific resources (you can add a function to do that if they are the same resources every time)

See the ResourceOffer operator as an example
https://github.com/liqotech/liqo/blob/master/pkg/liqo-controller-manager/resourceoffer-controller/resourceoffer_controller_test.go